src/main/java/teammates/ui/webapi/CompileLogsAction.java

```diff
@@ -20,47 +30,83 @@
 
-        long endTime = Instant.now().toEpochMilli();
+        Instant endTime = Instant.now();
         // Sets the range to 6 minutes to slightly overlap the 5 minute email timer
         long queryRange = 1000 * 60 * 6;
-        long startTime = endTime - queryRange;
+        Instant startTime = endTime.minusMillis(queryRange);
+
+        List<String> logOptions = Arrays.asList(
+                "resource.type=\"gae_app\"",
+                "resource.labels.module_id=\"default\"",
+                "logName=\"projects/" + options.getProjectId() + "/logs/appengine.googleapis.com%2Frequest_log\"",
+                "severity>=ERROR",
+                "timestamp>\"" + startTime.toString() + "\"",
+                "timestamp<=\"" + endTime.toString() + "\""
+        );
+
+        List<LogEntry> logs = new ArrayList<>();
+        Page<LogEntry> entries = logging.listLogEntries(
+                EntryListOption.filter(logOptions.stream().collect(Collectors.joining("\n")))
+        );
+
+        for (LogEntry logEntry : entries.iterateAll()) {
+            logs.add(logEntry);
+        }
+
+        return logs;
+    }
+
+    private void sendEmail(List<LogEntry> logs) {
+        List<String> logMessages = new ArrayList<>();
+        List<String> logLevels = new ArrayList<>();
+
+        for (LogEntry logEntry : logs) {
+            Any entry = (Any) logEntry.getPayload().getData();
 
-        LogQuery q = LogQuery.Builder.withDefaults()
-                                     .includeAppLogs(true)
-                                     .startTimeMillis(startTime)
-                                     .endTimeMillis(endTime)
-                                     .minLogLevel(LogLevel.ERROR);
+            JsonFormat.TypeRegistry tr = JsonFormat.TypeRegistry.newBuilder()
+                    .add(RequestLog.getDescriptor())
+                    .add(LogLine.getDescriptor())
+                    .add(SourceLocation.getDescriptor())
+                    .add(SourceReference.getDescriptor())
+                    .build();
 
-        Iterable<RequestLogs> logs = logService.fetch(q);
-        List<AppLogLine> errorLogs = new ArrayList<>();
+            List<LogLine> logLines = new ArrayList<>();
+            try {
+                String logContentAsJson = JsonFormat.printer().usingTypeRegistry(tr).print(entry);
 
-        for (RequestLogs requestLogs : logs) {
-            List<AppLogLine> logList = requestLogs.getAppLogLines();
+                RequestLog.Builder builder = RequestLog.newBuilder();
+                JsonFormat.parser().ignoringUnknownFields().usingTypeRegistry(tr).merge(logContentAsJson, builder);
+                RequestLog reconvertedLog = builder.build();
 
-            for (AppLogLine currentLog : logList) {
-                LogLevel logLevel = currentLog.getLogLevel();
+                logLines = reconvertedLog.getLineList();
+            } catch (InvalidProtocolBufferException e) {
+                // TODO
```

Comment:
What are we planning on doing here?

Reply:
@wkurniawan07 did you intend for this to be done in this PR or separately (i.e. needs some discussion)?

Reply:
I'm not sure myself what best to do here. If we do a error-level log here, then this log will be part of the next logs compilation, which will also fail because of the same reason. At best we can do a warn-level log here, but let's be real: that's as good as no logs.

Reply:
Yea I think this will need some discussion and possibly research, maybe settle another time with another PR/issue. We go ahead with what we have for now then